### PR TITLE
Return an iterator from enum_variant_ids.

### PR DESCRIPTION
**Pull request by Ms2ger**

### CODE REVIEW

File: src/librustc/metadata/decoder.rs [hunk 1]
```diff
@@ -268,10 +268,24 @@ fn item_trait_ref<'tcx>(doc: rbml::Doc, tcx: &ty::ctxt<'tcx>, cdata: Cmd)
...
+    cdata: Cmd<'a>,
+}
+
+impl<'a> Iterator for EnumVariantIds<'a> {
+    type Item = ast::DefId;
+
+    fn next(&mut self) -> Option<ast::DefId> {
+        self.iter.next().map(|p| translated_def_id(self.cdata, p))
+    }
+}
```
Comment (bluss) [inline src/librustc/metadata/decoder.rs]: Is it possible to use a better size_hint? It's important since the iterator is collected when used.

**Comment by Ms2ger** [comment]
Done

**Comment by bluss** [comment]
Thanks. I think the answer to my question should have been “no”, unfortunately. The underlying iterators have no size information and maybe won't have since it's kind of a filtering operation. Thanks anyway and this sets it up to improve if the adapted iterator improves, though.

**Pull request merged**